dbt/models/tax/schema.yml

```diff
@@ -23,7 +23,53 @@ sources:
 
       - name: pin
         description: '{{ doc("table_pin") }}'
-
+        columns:  
+          - name: tax_code_num
+            description: '{{ doc("shared_column_tax_code")}}'
+            tests:
+              - row_values_match_after_join:
+                  name: pin_tax_dist_matches_iasworld_legdat
+                  external_model: source('iasworld', 'legdat')
+                  external_column_name: taxdist
+                  column_alias: tax_code_num
+                  external_column_alias: taxdist
+                  group_by: 
+                    - year
+                    - pin
+                  join_condition:
+                    ON  model.pin = external_model.parid
+                    AND model.year = external_model.taxyr
+                    AND external_model.cur = 'Y'
+                    AND external_model.deactivat IS NULL
+                  config:
+                    where:
+                      CAST(year AS int) BETWEEN {{ var('test_qc_year_start') }} AND {{ var('test_qc_year_end') }}
```

Comment:
**[Nitpick, non-blocking]** For completeness' sake, let's add the `meta.description` attribute to this test so that we can properly output it for review if it starts failing:

```suggestion
                  config:
                    where:
                      CAST(year AS int) BETWEEN {{ var('test_qc_year_start') }} AND {{ var('test_qc_year_end') }}
                  meta:
                    description: tax_code_num should match legdat.taxdist
```